e2e-tests/tests/test_image_redactor.py

```diff
@@ -0,0 +1,44 @@
+import os
+
+from common.methods import image_redactor
+
+
+def test_given_image_with_int_color_filling_then_we_get_redact_image():
+    # gray redact
+    expected_result_image = get_file("gray_redact.png")
+    response = image_redactor("original_image.png", 255)
+    validate_and_close(expected_result_image, response)
+
+
+def test_given_image_with_tuple_color_filling_then_we_get_redact_image():
+    # purple redact
+    expected_result_image = get_file("purple_redact.png")
+    response = image_redactor("original_image.png", "102, 0, 102")
+    validate_and_close(expected_result_image, response)
+
+
+def test_given_image_with_no_color_filling_then_we_get_redact_image():
+    # black redact
+    expected_result_image = get_file("black_redact.png")
+    response = image_redactor("original_image.png")
+    validate_and_close(expected_result_image, response)
+
+
+def test_given_no_image_then_we_fail():
+    # black redact
+    expected_response = '{"error": "Invalid parameter, please insert image data"}'
+    response = image_redactor("")
+    assert response.status_code == 422
+    assert response.content.decode() == expected_response
+
+
+def get_file(file_name: str):
+    current_dir = os.path.dirname(__file__)
+    file_path = os.path.join(current_dir, "..", "resources", file_name)
+    return open(file_path, "rb")
```

Comment:
You can use the same method here for the request, and pass the content to the 'methods utils', making it just handle the methods and not read from files, avoiding duplication as well.

Reply:
Also if we know that the image length in bytes is not too large and stable on running the tests locally and in CI, you can read the content and close the file in this method, using 'with', and not depend on other method closing the file (than one might forget when adding a new scenario that's not using validate_and_close)

Reply:
Done.

Reply:
The image can be large... so not sure we should be duplicating the bytes, wdyt?

Reply:
Not a must

Reply:
if its just for this test, use a small image, but really not a must